src/lib/reasoners/satml_frontend.ml

```diff
@@ -1350,6 +1357,10 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
            (* don't attempt to compute a model if timeout before
               calling unsat function *)
            i_dont_know env (Timeout Assume)
+         | Util.Step_limit_reached _ ->
+           (* When reaching the step limit on an assume, we do not want to
+              answer 'unknown' right away. *)
+           env
```

Comment:
Same as above

Reply:
By blocking the instructions at the frontend level, we ensure the environment cannot be used in a wrong way. Besides, we may want to display information from the invalid environment when the step limit is reached, so I would rather keep it like that.